mlos_bench/mlos_bench/optimizers/base_optimizer.py

```diff
@@ -269,13 +267,15 @@ def suggest(self) -> TunableGroups:
             These are the same tunables we pass to the constructor,
             but with the values set to the next suggestion.
         """
+        self._iter += 1
```

Comment:
Here's another alternative:
1. In `suggest`, place the config in a `pending` set.
  - As an extension on this, we could also internally loop on `_suggest` until we received a config not in the `pending` set, but I'd leave that for future work to explore.
1. In `register`, check to see if the provided configs are in the `pending` set.
  - If they are, remove them from the set and increment the iteration count.
  - Finish registering the config-scores regardless.

This allows the caller to "throw out" some "bad" configs (which they may filter with their own logic) but avoid those configs counting against them.

We have an API that's sort of for this in `mlos_core`: `register_pending()`
The basic idea was to allow us some means of avoiding the optimizer repeatedly suggesting the same config to us if we had some other means of identifying it as invalid or otherwise "pending" (e.g., doing parallel trials with different configs).

Obviously in the perfect world we'd also provide feedback to the optimizer about a config being no good (e.g., register it as 3x worst real config-score seen so far), but this allows us still a sane middle ground approach.

Reply:
yeah, we can do that, but I don't think this extra functionality belongs to the base optimizer class. I would rather implement it in some derived class to keep the base interface simpler and also give the user more choice when implementing/instantiating the optimizers.

P.S. On a side note, I would be very careful when tracking the suggested/pending configurations. Exact matching might not work - e.g., consider two configurations that differ by 1e-6 in one parameter. Ideally, the user should be able to pick some similarity measure and closeness boundary for their configurations.

P.P.S. Same applies for termination criteria - e.g., we might want to stop the optimization not only when the acquisition function is small on the entire range, but also when, say, teh last 10 suggested configs are all close to the already tried configs, etc., etc.

Reply:
Termination criteria I think could be implemented separately, but your point about the first one is somewhat well taken.  However, part of me thinks that most of our configs are not continuous so that particular scenario would only show up except for floating point precision issues between numpy and native which we should mask.

Reply:
I guess the main behavior I still somewhat prefer is that the count increases not when I call suggest, but when I call register, especially since I may wish to schedule that config at some point in the future and don't want to see it suggested again in the meantime.
But again, that count tracking should be independent of the number of times we repeat them.

Reply:
That said, I'm not certain I'm super strong on any of these approaches.  I think they all have pros and cons.

Reply:
> that particular scenario would only show up except for floating point precision issues between numpy and native which we should mask.

oh no, it will be everywhere - e.g. for `sched_migration_cost_ns` (an integer, by the way) - are the values of 500000 and 500100 different or the same? how about 500000 and 500001? etc., etc.

Reply:
I would argue those quantization issues are application domain specfic, and probably quantization specification issues in that case and are in fact different configs from the optimizer perspective.

Reply:
exactly. For me, that's another argument for factoring out the tracking of suggested/pending configs etc. into a separate class and PR